pkg/worker/worker_test.go

```diff
@@ -0,0 +1,119 @@
+package worker
+
+import (
+	"slices"
+	"testing"
+	"time"
+
+	"github.com/0xSplits/specta/pkg/envvar"
+	"github.com/0xSplits/specta/pkg/recorder"
+	"github.com/0xSplits/specta/pkg/runtime"
+	"github.com/0xSplits/specta/pkg/worker/handler"
+	"github.com/xh3b4sd/logger"
+)
+
+func Test_Worker_handler_execution(t *testing.T) {
```

Comment:
This is a pretty cool test to verify the failure isolation. In CI we run those tests with the race detector.